src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/MembersMustExist.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Microsoft.DotNet.ApiCompatibility.Extensions;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// Rule that evaluates whether a member exists on Left and Right.
+    /// If the member doesn't exist on Right but it does on Left, it adds a <see cref="CompatDifference"/> to the list of differences.
+    /// </summary>
+    public class MembersMustExist : Rule
+    {
+        /// <summary>
+        /// Evaluates whether a type exists on both sides of the <see cref="TypeMapper"/>.
+        /// </summary>
+        /// <param name="mapper">The <see cref="TypeMapper"/> to evaluate.</param>
+        /// <param name="differences">The list of <see cref="CompatDifference"/> to add differences to.</param>
+        public override void Run(TypeMapper mapper, IList<CompatDifference> differences)
+        {
+            ITypeSymbol left = mapper.Left;
+            if (left != null && mapper.Right == null)
+                differences.Add(new CompatDifference(DiagnosticIds.TypeMustExist, $"Type '{left.ToDisplayString()}' exists on the contract but not on the implementation", DifferenceType.Removed, left));
```

Comment:
are there any good alternatives to contract and implementation ? 
we will be using this for a lot more purposes as well.

Reply:
Yes. That is on one of my cards. I will add a way to plugin this through settings (how to name left and right). Maybe we can default them to left and right?

Reply:
yeah left and right sounds better

Reply:
It's also what we have in our parameter naming guidelines: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-parameters
> ✔️ DO use left and right for binary operator overload parameter names if there is no meaning to the parameters.